osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/BatchInserter.cs

```diff
@@ -182,10 +169,24 @@ private async Task run(HighScore[] scores)
             if (insertCount == 0)
             {
                 Console.WriteLine($" Skipped all {scores.Length} scores");
-
                 return;
             }
 
+            StringBuilder insertBuilder = new StringBuilder(
+                "INSERT INTO scores (`user_id`, `ruleset_id`, `beatmap_id`, `has_replay`, `preserve`, `ranked`, `rank`, `passed`, `accuracy`, `max_combo`, `total_score`, `data`, `pp`, `legacy_score_id`, `legacy_total_score`, `ended_at`, `unix_updated_at`) VALUES ");
+
+            foreach (var score in scores)
+            {
+                if (string.IsNullOrEmpty(score.InsertSql))
+                    continue;
+
+                if (!first)
```

Comment:
I had to scroll up to the veeeeeeeeeeeeeeeeeeery start of this method, through parallelised code, to find this variable. Please move it closer to here in your next changes so I can be more assured that there isn't some weird threading thing going on.